test/node-test/diagnostics-channel/fetch.js

```diff
@@ -0,0 +1,115 @@
+const { createServer } = require('http')
+
+describe('diagnosticsChannel for fetch', () => {
+  let server
+  before(() => {
+    server = createServer((req, res) => {
+      res.setHeader('Content-Type', 'text/plain')
+      res.setHeader('trailer', 'foo')
+      res.write('hello')
+      res.addTrailers({
+        foo: 'oof'
+      })
+      res.end()
+    })
+  })
+
+  after(() => { server.close() })
+
+  test('fetch', async t => {
+    t = tspl(t, { plan: 42 })
+
+    let startCalled = 0
+
+    diagnosticsChannel.channel('undici:fetch:start').subscribe(({ input, init }) => {
+      startCalled += 1
+      if (init.redirect) {
+        t.strictEqual(input, 'badrequest')
+        t.deepStrictEqual(init, { redirect: 'error' })
+      } else {
+        t.strictEqual(input, `http://localhost:${server.address().port}`)
+        t.deepStrictEqual(init, {})
+      }
+    })
+
+    let endCalled = 0
+    diagnosticsChannel.channel('undici:fetch:end').subscribe(({ input, init, result, error }) => {
+      endCalled += 1
+      if (init.redirect) {
+        t.strictEqual(input, 'badrequest')
+        t.deepStrictEqual(init, { redirect: 'error' })
+      } else {
+        t.strictEqual(input, `http://localhost:${server.address().port}`)
+        t.deepStrictEqual(init, {})
+      }
+      t.strictEqual(result, null)
+    })
+
+    let asyncStartCalled = 0
+    diagnosticsChannel.channel('undici:fetch:asyncStart').subscribe(({ input, init, result, error }) => {
+      asyncStartCalled += 1
+      if (init.redirect) {
+        t.strictEqual(input, 'badrequest')
+        t.deepStrictEqual(init, { redirect: 'error' })
+      } else {
+        t.strictEqual(input, `http://localhost:${server.address().port}`)
+        t.deepStrictEqual(init, {})
+        t.ok(result)
+      }
+    })
+
+    let asyncEndCalled = 0
+    diagnosticsChannel.channel('undici:fetch:asyncEnd').subscribe(async ({ input, init, result, error }) => {
+      asyncEndCalled += 1
+      if (init.redirect) {
+        t.strictEqual(input, 'badrequest')
+        t.deepStrictEqual(init, { redirect: 'error' })
+        t.strictEqual(result, null)
+        t.ok(error)
+        t.strictEqual(error.cause.code, 'ERR_INVALID_URL')
+      } else {
+        t.strictEqual(input, `http://localhost:${server.address().port}`)
+        t.deepStrictEqual(init, {})
+        t.ok(result)
+        t.strictEqual(result.status, 200)
+        t.strictEqual(error, null)
+      }
+    })
+
+    let errorCalled = 0
+    diagnosticsChannel.channel('undici:fetch:fetchError').subscribe(async ({ input, init, error }) => {
+      errorCalled += 1
+      if (init.redirect) {
+        t.strictEqual(input, 'badrequest')
+        t.deepStrictEqual(init, { redirect: 'error' })
+        t.ok(error)
+        t.strictEqual(error.cause.code, 'ERR_INVALID_URL')
+      } else {
+        t.strictEqual(input, `http://localhost:${server.address().port}`)
+        t.deepStrictEqual(init, {})
+        t.notOk(error)
```

Comment:
actually we never hit this. 

```suggestion
        t.ok(error != undefined)
```